### PR TITLE
Import/export COCO categories

### PR DESCRIPTION
**Pull request by brimoor**
Closes https://github.com/voxel51/fiftyone/issues/500.

The `COCODetectionDataset` format will now load the `categories` dict from COCO labels and use it (if present) to populate a `supercategory` attribute on each detection.

The `categories` field of the COCO labels is stored in the dataset's `info` field, so that this information can be included (if available) when exporting datasets in `COCODetectionDataset` format.

Test code:

```py
import fiftyone as fo
import fiftyone.zoo as foz

export_dir1 = "/tmp/coco-export1"
export_dir2 = "/tmp/coco-export2"

#
# Load a dataset
#
# Ignore the fact that this dataset is itself COCO, since the dataset zoo does not
# currently store supercategory info (all detection datasets are currently stored in
# FiftyOneImageDetectionDataset format, which doesn't support storing this info)
#
dataset = foz.load_zoo_dataset("coco-2017", split="validation")

#
# Export one sample in COCO format
# `dataset` does not have `categories` in its `info` field, so the output dataset will not
# have supercategories populated
#
dataset.limit(1).export(export_dir1, fo.types.COCODetectionDataset)

#
# Load the exported dataset. The export did not have supercategories (they are `None`), so
# the imported detections will not have `supercategory` attributes
#
dataset1 = fo.Dataset.from_dir(export_dir1, fo.types.COCODetectionDataset)

print(dataset1.first())
```

```
<Sample: {
    'id': '5f50fbb688e56149fab02ce4',
    'filepath': '/tmp/coco-export1/data/000001.jpg',
    'tags': BaseList([]),
    'metadata': <ImageMetadata: {
        'size_bytes': None,
        'mime_type': None,
        'width': 640,
        'height': 426,
        'num_channels': None,
    }>,
    'ground_truth': <Detections: {
        'detections': BaseList([
            <Detection: {
                'id': '5f50fbb688e56149fab02cd0',
                'label': 'potted plant',
                'bounding_box': array([0.3703125 , 0.33450704, 0.03859375, 0.16314554]),
                'confidence': None,
                'attributes': BaseDict({
                    'area': <NumericAttribute: {'value': 531.8071000000001}>,
                    'iscrowd': <NumericAttribute: {'value': 0.0}>,
                }),
            }>,
            ...
        ]),
    }>,
}>
```

```py
# Populate supercategorires
for category in dataset1.info["categories"]:
    category["supercategory"] = "object"
dataset1.save()

#
# Export the dataset in COCO format again, this time with supercategories
#
dataset1.export(export_dir2, fo.types.COCODetectionDataset)

#
# Load the exported dataset with supercategories. This time the detections do have
# supercategory attributes populated
#
dataset2 = fo.Dataset.from_dir(export_dir2, fo.types.COCODetectionDataset)
print(dataset2.first())
```

```
<Sample: {
    'id': '5f50fbb688e56149fab02d22',
    'filepath': '/tmp/coco-export2/data/000001.jpg',
    'tags': BaseList([]),
    'metadata': <ImageMetadata: {
        'size_bytes': None,
        'mime_type': None,
        'width': 640,
        'height': 426,
        'num_channels': None,
    }>,
    'ground_truth': <Detections: {
        'detections': BaseList([
            <Detection: {
                'id': '5f50fbb688e56149fab02d0e',
                'label': 'potted plant',
                'bounding_box': array([0.3703125 , 0.33450704, 0.03859375, 0.16314554]),
                'confidence': None,
                'attributes': BaseDict({
                    'supercategory': <CategoricalAttribute: {'value': 'object', 'confidence': None, 'logits': None}>,
                    'area': <NumericAttribute: {'value': 531.8071000000001}>,
                    'iscrowd': <NumericAttribute: {'value': 0.0}>,
                }),
            }>,
            ...
        ]),
    }>,
}>
```

### CODE REVIEW

**Review by ehofesmann — APPROVED** [review]
This works well for importing custom datasets and should be merged ASAP to respond to Martin.

I still think that it is pretty important to get supercategories when loading COCO from the zoo as well though. We really don't want to be throwing out information.
